torch/_dynamo/backends/debugging.py

```diff
@@ -113,3 +117,18 @@ def relu_accuracy_error_TESTING_ONLY(gm: torch.fx.GraphModule, example_inputs):
     gm.recompile()
 
     return gm
+
+
+@register_backend
+def non_leaf_compile_error_TESTING_ONLY(gm: torch.fx.GraphModule, example_inputs):
+    # Require at least one non-trivial thing in the graph,
+    # see https://github.com/pytorch/pytorch/issues/102898
+    for node in gm.graph.nodes:
+        if node.op == "call_function":
+            break
+    else:
+        return gm
+    for t in example_inputs:
+        if t.grad_fn is not None:
```

Comment:
nit to keep in line with PR title :p 
```suggestion
        if not t.is_leaf:
```